pyp2rpm/package_getters.py

```diff
@@ -126,21 +127,27 @@ class PypiDownloader(PackageGetter):
 
     """Class for downloading the package from PyPI."""
 
-    def __init__(self, client, name, version=None, save_dir=None):
+    def __init__(self, client, name, version=None, prerelease=False,
+                 save_dir=None):
         self.client = client
         self.name = name
         try:
-            self.versions = self.client.package_releases(self.name)
+            self.versions = self.client.package_releases(self.name, True)
         except xmlrpclib.ProtocolError as e:
             sys.stderr.write("Failed to connect to server: {0} \n".format(e))
             raise SystemExit(3)
 
+        # Use only stable versions, unless --pre was specified
+        if not prerelease:
+            self.versions = [candidate for candidate in self.versions
+                             if not parse_version(candidate).is_prerelease]
+
         # If versions is empty list then there is no such package on PyPI
         if not self.versions:
             raise exceptions.NoSuchPackageException(
                 'Package "{0}" could not be found on PyPI.'.format(name))
 
-        self.version = version or self.versions[-1]
+        self.version = version or self.versions[0]
```

Comment:
Why are you using the first instead of the last version here?
Is the added `True` flag above sorting from latest to oldest?

Reply:
https://warehouse.pypa.io/api-reference/xml-rpc/
The True flag gets all releases, while only the latest release is returned otherwise.  The latest release might be a prerelease, so we need the full list if we want find the latest stable release.  I don't know how pypi behaved when "-1" was used, but I assume that the API used to return the full list by default, sorted in the opposite order.

I can add a comment to that effect.

```
>>> import xmlrpc.client as xmlrpclib
>>> client = xmlrpclib.ServerProxy('https://pypi.org/pypi')
>>> client.package_releases('pyp2rpm', True)
['3.3.3', '3.3.2', '3.3.1', '3.3.0', '3.2.3', '3.2.2', '3.2.1', '3.2.0', '3.1.3', '3.1.2', '3.1.1', '3.1.0', '3.0.2', '3.0.1', '3.0.0', '2.0.0', '1.1.2', '1.1.1', '1.1.0b', '1.1.0a', '1.0.1', '1.0.0', '0.5.2', '0.5.1', '0.5.0', '0.4.2', '0.4.1', '0.4.0', '0.3.1', '0.3.0', '0.2.0', '0.1.0']
>>> client.package_releases('pyp2rpm')
['3.3.3']
```

Reply:
Great, then that's fine.